### DISCUSSION TITLE
How to save result of single computation without `parallelReduce`?

### DISCUSSION THREAD

**Discussion by MetaB0y** [reach-sh/reach-lang#1184] [Q&A] [answered]
I have the following pattern quite often:

I am waiting for somebody to do some action via `api` call (because Participants can't reconnect). I want to do some computation and save the result to reuse later in the program.

I am doing something like
```
  const [running, result] = parallelReduce([true, 0])
    .invariant(true)
    .while(running)
    .api_(Api.myApi, () => {
      check(...);

      return [
        (arg, callback) => {
          callback(null);
          
          return [false, computeSomthing(arg)];
        },
      ];
    })
```

It works fine but the problem is that it uses `parallelReduce` which should not be necessary here because I am doing an action only once. I think it should be `fork` but I am not sure how to do this with `fork`.

**Comment by DanBurton** [comment, marked as answer]
I believe you are looking for `call`, which is the basic way to indicate that a specific API must be called to advance the program.

https://docs.reach.sh/rsh/step/#p_78

```
const [ [arg], callback ] =
  call(Api.myApi)
    .check((arg) => {
      check(...);
    });
callback(null);
const result = computeSomthing(arg);
commit();
```
